src/libpgagroal/prometheus.c

```diff
@@ -136,273 +136,323 @@ pgagroal_prometheus(int client_fd)
    exit(1);
 }
 
+int
+pgagroal_init_prometheus(void * shm)
+{
+   struct prometheus* prometheus;
+   
+   prometheus = (struct prometheus*)shm;
+
+   for (int i = 0; i < HISTOGRAM_BUCKETS; i++)
+   {
+      atomic_init(&prometheus->session_time[i], 0);
+   }
+   atomic_init(&prometheus->session_time_sum, 0);
+
+   atomic_init(&prometheus->connection_error, 0);
+   atomic_init(&prometheus->connection_kill, 0);
+   atomic_init(&prometheus->connection_remove, 0);
+   atomic_init(&prometheus->connection_timeout, 0);
+   atomic_init(&prometheus->connection_return, 0);
+   atomic_init(&prometheus->connection_invalid, 0);
+   atomic_init(&prometheus->connection_get, 0);
+   atomic_init(&prometheus->connection_idletimeout, 0);
+   atomic_init(&prometheus->connection_flush, 0);
+   atomic_init(&prometheus->connection_success, 0);
+
+   atomic_init(&prometheus->auth_user_success, 0);
+   atomic_init(&prometheus->auth_user_bad_password, 0);
+   atomic_init(&prometheus->auth_user_error, 0);
+
+   for (int i = 0; i < NUMBER_OF_SERVERS; i++)
+   {
+      atomic_init(&prometheus->server_error[i], 0);
+   }
+   atomic_init(&prometheus->failed_servers, 0);
+
+   struct configuration* config = NULL;
+   config = (struct configuration*) shmem;
+
+   for (int i = 0; i < config->max_connections; i++)
+   {
+      memset(prometheus->prometheus_connections[i].database, 0, MAX_DATABASE_LENGTH);
```

Comment:
Remove this `memset` with a general `memset(&prometheus->prometheus_connections[i], 0, sizeof(struct prometheus_connection);`